src/main/java/com/cognifide/repository/BookRepositoryInMemory.java

```diff
@@ -0,0 +1,74 @@
+package com.cognifide.repository;
+
+import com.cognifide.domain.Book;
+import com.cognifide.domain.BookNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by cognifide.nowak on 04.04.2016.
+ */
+@Component
+public class BookRepositoryInMemory implements BookRepository {
+
+    private List<Book> bookList = Collections.synchronizedList(new ArrayList<Book>());
+
+
+    public Book findOne(int id) {
+        try {
+            return bookList.get(id);
+        } catch (IndexOutOfBoundsException e) {
+            throw new BookNotFoundException();
+        }
+    }
+
+    public void save(Book book) {
+        bookList.add(book);
+    }
+
+    public Book delete(int id) {
+        try {
+            Book temp = bookList.remove(id);
+            return temp;
+        } catch (IndexOutOfBoundsException e ) {
+            throw new BookNotFoundException();
+        }
+    }
+
+    public List<Book> findByAuthor(String author) {
+        List<Book> results = new ArrayList<Book>();
+        for (Book book : bookList) {
+            if(book.getAuthor().equals(author))
+                results.add(book);
+        }
+        if(results.size() == 0)
```

Comment:
`if (booleanValue())` should have a space between `if` and parenthesis, otherwise `for` and `if` formatting are inconsistent.